src/conf_mode/interfaces_openvpn.py

```diff
@@ -487,6 +507,25 @@ def verify(openvpn):
     # not depending on any operation mode
     #
 
+    # verify that local_host/remote_host match with any ip_version override
+    # specified (if a dns name is specified for remote_host, no attempt is made
+    # to verify that record resolves to an address of the configured family)
+    if 'ip_version' in openvpn:
+        if 'local_host' in openvpn:
+            if openvpn['ip_version'] == 'ipv4' and is_ipv6(openvpn['local_host']):
+                raise ConfigError('Cannot use an IPv6 "local-host" with protocol-version ipv4')
+            elif openvpn['ip_version'] == 'ipv6' and is_ipv4(openvpn['local_host']):
+                raise ConfigError('Cannot use an IPv4 "local-host" with protocol-version ipv6')
+
+        if 'remote_host' in openvpn:
+            remote_hosts = dict_search('remote_host', openvpn)
+            for remote_host in remote_hosts:
+                if openvpn['ip_version'] == 'ipv4' and is_ipv6(remote_host):
+                    raise ConfigError('Cannot use an IPv6 "remote-host" with protocol-version ipv4')
```

Comment:
Nitpick: why `"remote-host"` is quoted but `protocol-version` is not? We don't have a policy about quoting option names, but I believe we should be consistent about it within a single message. I'd probably quote `"protocol-version ipv4"`.